seatunnel-core/seatunnel-starter/src/main/java/org/apache/seatunnel/core/starter/seatunnel/command/ClientExecuteCommand.java

```diff
@@ -112,14 +116,24 @@ public void execute() throws CommandExecuteException {
                 startTime = LocalDateTime.now();
                 // create job proxy
                 ClientJobProxy clientJobProxy = jobExecutionEnv.execute();
+                // register cancelJob hook
+                if (clientCommandArgs.isCloseJob()) {
+                    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+                        try {
+                            shutdownHook(clientJobProxy);
+                        } catch (Exception e) {
+                            throw new SeaTunnelEngineException(String.format("Cancel job exception and error is %s", ExceptionUtils.getMessage(e)));
```

Comment:
need add log.error in here. Throw exception useless.

Reply:
> need add log.error in here. Throw exception useless.

I see.